### ISSUE TITLE
PostgresDB for Keycloak Helm Chart Crashing

### ISSUE THREAD

**Issue by sahilmalav** [bitnami/charts#10736]
### Name and Version

bitnami/keycloak 6.2.5

### What steps will reproduce the bug?

I had the Bitnami Keycloak Helm chart (version 7.1.18) running on my cluster, when suddenly it started crashing. The initial error was due to the readiness probe being failed on the postgres deployment. I tried reinstalling it but the error didn't go away. I thought that the error might be due to some database files being locked, so I reinstalled the chart with a different mount location (so that I get a clean slate for setting up postgres). That didn't help either. I have another instance of Keycloak chart running version 6.2.5, so I downgraded it to that version. But I get the same error still. Below is the error that I see in the postgres pod: 
```
postgresql 05:18:59.57 
postgresql 05:18:59.57 Welcome to the Bitnami postgresql container
postgresql 05:18:59.58 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql
postgresql 05:18:59.58 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql/issues
postgresql 05:18:59.58 
postgresql 05:18:59.60 INFO  ==> ** Starting PostgreSQL setup **
postgresql 05:18:59.64 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql 05:18:59.65 INFO  ==> Loading custom pre-init scripts...
postgresql 05:18:59.65 INFO  ==> Initializing PostgreSQL database...
postgresql 05:18:59.72 INFO  ==> Cleaning stale /bitnami/postgresql/data/postmaster.pid file
postgresql 05:19:00.21 INFO  ==> pg_hba.conf file not detected. Generating it...
postgresql 05:19:00.21 INFO  ==> Generating local authentication configuration
postgresql 05:19:00.29 INFO  ==> Deploying PostgreSQL with persisted data...
postgresql 05:19:00.30 INFO  ==> Configuring replication parameters
postgresql 05:19:00.34 INFO  ==> Configuring fsync
postgresql 05:19:00.35 INFO  ==> Configuring synchronous_replication
postgresql 05:19:00.73 INFO  ==> Loading custom scripts...
postgresql 05:19:00.74 INFO  ==> Enabling remote connections
postgresql 05:19:00.76 INFO  ==> ** PostgreSQL setup finished! **
postgresql 05:19:00.79 INFO  ==> ** Starting PostgreSQL **
2022-06-14 05:19:02.027 GMT [1] LOG:  pgaudit extension initialized
2022-06-14 05:19:02.028 GMT [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2022-06-14 05:19:02.028 GMT [1] LOG:  listening on IPv6 address "::", port 5432
2022-06-14 05:19:02.885 GMT [1] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2022-06-14 05:19:07.359 GMT [93] LOG:  database system shutdown was interrupted; last known up at 2022-06-14 05:02:56 GMT
2022-06-14 05:19:10.159 GMT [99] FATAL:  the database system is starting up
2022-06-14 05:19:20.169 GMT [105] FATAL:  the database system is starting up
2022-06-14 05:19:29.209 GMT [111] FATAL:  the database system is starting up
2022-06-14 05:19:30.156 GMT [117] FATAL:  the database system is starting up
2022-06-14 05:19:39.214 GMT [123] FATAL:  the database system is starting up
```
and below is the error I see in the keycloak pod (I assume this is due to postgres not being up): 
```
keycloak 05:27:20.66 
keycloak 05:27:20.67 Welcome to the Bitnami keycloak container
keycloak 05:27:20.67 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-keycloak
keycloak 05:27:20.67 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-keycloak/issues
keycloak 05:27:20.67 
keycloak 05:27:20.67 INFO  ==> ** Starting keycloak setup **
keycloak 05:27:20.69 INFO  ==> Validating settings in KEYCLOAK_* env vars...
keycloak 05:27:20.70 INFO  ==> Trying to connect to PostgreSQL server my-release-postgresql...
timeout reached before the port went into state "inuse"
timeout reached before the port went into state "inuse"
```
I've tried everything I could think of, but I still can't get postgres pod to work. Can someone please help me out here and provide suggestions to resolve this issue? 

### Are you using any custom parameters or values?

`--set auth.adminUser=admin --set auth.adminPassword=admin`

### What is the expected behavior?

Postgres pod should be up and running

### What do you see instead?

Postgres (and consequently Keycloak) pods are crashing.

**Comment by carrodher** [comment]
I'm not able to reproduce the issue from a fresh deployment. It seems it is not an issue related to the Bitnami keycloak container image or Helm chart but about how the application or environment is being used/configured.

You can try adding the `postgresql.image.debug=true` parameter to see more logs during the PostgreSQL container initialization.

For information regarding the application itself, customization of the content within the application, or questions about the use of technology or infrastructure; we highly recommend checking forums and user guides made available by the project behind the application or the technology.

That said, we will keep this ticket open until the stale bot closes it just in case someone from the community adds some valuable info.

**Comment by sahilmalav** [comment]
@carrodher Thanks for taking the time out to dig into it. I was finally able to figure it out after a couple of days of debugging. You were correct - it wasn't an issue related to the chart. As it turns out, my NFS started performing poorly all of a sudden (not sure why, still digging into that). I had mounted the PostgreSQL pod on the NFS. So it wasn't able to create the init files there due to the poor performance of NFS. Once I changed the mount to my local disk, it started working. Closing this issue now. Thanks again!
